models/errors.go

```diff
@@ -15,15 +15,16 @@ func (err simpleError) Error() string {
 	return err.message
 }
 
+// NewSimpleError returns the custom defined error of type SimpleError.
 func NewSimpleError(msg string) simpleError {
```

Comment:
I don't really understand why we need to export SimpleError at all. It is meant as an implementation device for the other classes, not to be ever created.